cmd/all-in-one/Dockerfile

```diff
@@ -4,8 +4,16 @@ RUN apk add build-base
 
 WORKDIR /app
 
-# COPY All things inside the project and build
-COPY . .
+# COPY relevant files and build
+COPY go.mod .
+COPY go.sum .
+COPY cmd/all-in-one/ ./cmd/all-in-one/
+COPY teletrace-otelcol/ ./teletrace-otelcol
+COPY model/ ./model
+COPY web ./web
```

Comment:
Any reason why we need to copy the entire frontend?

Reply:
@ophiryael I took the dirs from https://github.com/teletrace/teletrace/blob/c58a2173b0969f92844bc28e7db8301e563e8dd5/.goreleaser.yml#L78

Is the frontend not a part from this docker?

Reply:
We are using a [multi-stage Dockerfile](https://docs.docker.com/build/building/multi-stage/) in the following way:
1. Build the backend.
2. Build the frontend.
3. Combine them into a single image under different directories.

The `web` directory is not necessary for the backend build. It is relevant only for the second stage - the frontend build. Which has its own copy instruction: `COPY web/ .`. Therefore there is no need also to copy it in the first stage.

Regarding the GoReleaser, it first creates a temporary directory and links all the files under `extra_files`, and only then runs the Dockerfile build. So it has to copy the `web` directory beforehand so the frontend build stage won't fail.
@maorlx, please validate that I'm not writing nonsense about the goreleaser.

Reply:
Agreed with the first part - the frontend build has its own copy instruction, therefore its indeed not needed.

As @ophiryael  said, the GoReleaser just prepare all the relevant files/dirs which should be "accessed" by the entire Dockerfile (without extra_files section, the Dockerfile would have access only to Teletrace all-in-one binary file). So I would say its irrelevant to the copy instructions (different unrelated flow)